src/plugins/telemetry/public/services/telemetry_service.ts

```diff
@@ -156,30 +163,33 @@ export class TelemetryService {
   };
 
   public fetchLastReported = async (): Promise<number | undefined> => {
-    const response = await this.http.get<{ lastReported?: number }>(
-      '/api/telemetry/v2/last_reported'
+    const response = await this.http.get<FetchLastReportedResponse>(
+      LastReportedRoute,
+      INTERNAL_VERSION
     );
     return response?.lastReported;
   };
 
   public updateLastReported = async (): Promise<number | undefined> => {
-    return this.http.put('/api/telemetry/v2/last_reported');
+    return this.http.put(LastReportedRoute);
   };
 
-  /** Fetches an unencrypted telemetry payload so we can show it to the user **/
+  /** Fetches an unencrypted telemetry payload, so we can show it to the user **/
   public fetchExample = async (): Promise<UnencryptedTelemetryPayload> => {
     return await this.fetchTelemetry({ unencrypted: true, refreshCache: true });
   };
 
   /**
    * Fetches telemetry payload
    * @param unencrypted Default `false`. Whether the returned payload should be encrypted or not.
+   * @param refreshCache Default `false`. Set to `true` to force the regeneration of the telemetry report.
    */
   public fetchTelemetry = async <T = EncryptedTelemetryPayload | UnencryptedTelemetryPayload>({
     unencrypted = false,
     refreshCache = false,
   } = {}): Promise<T> => {
-    return this.http.post('/api/telemetry/v2/clusters/_stats', {
+    return this.http.post(FetchSnapshotTelemetry, {
+      ...INTERNAL_VERSION,
```

Comment:
This code looks good, but it does make me wonder whether we might consider providing a away to make a "version scoped" client so that you'd have smth like:

```ts
const httpV1 = this.http.createScopedClient({ version: '1' })
...
httpV1.post(FetchSnapshotTelemetry...)
```

but yeah, just a thought, nothing to action right now.

Reply:
That's a great idea! I would ask contributors if they'd use it.